packages/es-components/src/components/patterns/datepicker/DatePicker.js

```diff
@@ -29,36 +29,65 @@ export const DatePicker = props => {
   const {
     additionalHelpContent,
     children,
+    excludeDates,
+    filterDate,
+    highlightDates,
+    includeDates,
     labelText,
+    name,
+    onChange,
+    onChangeRaw,
+    onBlur,
     placeholder,
     selectedDate,
+    selectsEnd,
+    selectsStart,
+    startDate,
+    endDate,
     theme,
     ...otherProps
   } = props;
 
+  /* eslint-disable react/forbid-foreign-prop-types */
+  const textboxProps = omit(otherProps, ReactDatePicker.propTypes);
+  const datepickerProps = pick(otherProps, ReactDatePicker.propTypes);
```

Comment:
Would it be possible to use _just_ this, rather than declaring all the other props manually?

Reply:
Yes, that would clean up the internals of the component nicely. I was originally thinking it was nice to have the props we are primarily using explicitly set but as long as they're in propTypes I think that's good enough. 

And now that I have made that change I can see my omit/pick is not working quite how I expected, so I'm glad you had me try that! Don't merge yet.

Reply:
Fixed.